src/sentry/ingest/billing_metrics_consumer.py

```diff
@@ -108,35 +121,176 @@ def __init__(
         self,
         commit: Callable[[Mapping[Partition, Position]], None],
         max_batch_size: int,
-        max_batch_time: int,
     ) -> None:
-        self.__commit = commit
-        self.__max_batch_size = max_batch_size
-        self.__max_batch_time = timedelta(milliseconds=max_batch_time)
-        self.__messages_since_last_commit = 0
-        self.__last_commit = datetime.now()
-        self.__ready_to_commit: MutableMapping[Partition, Position] = {}
-        self.__closed = False
+        self._closed: bool = False
+        self._commit = commit
+        # XXX(iker): these max* variables currently don't behave as maximums,
+        # but rather minimums. We should fix this at some point.
+        self._max_batch_size: int = max_batch_size
+
+        self._billing_topic = Topic(settings.KAFKA_OUTCOMES_BILLING)
+        self._billing_producer = self._get_billing_producer()
+        # Especially on incidents generating big backlogs, we must expect a lot
+        # of removals from the beginning of a big FIFO. A dequeue provides O(1)
+        # time on removing items from the beginning; while a regular list takes
+        # O(n).
+        self._ongoing_billing_outcomes: Deque[MetricsBucketBilling] = deque()
+        self._ready_to_commit: MutableMapping[Partition, Position] = {}
+        self._messages_ready_since_last_commit: int = 0
```

Comment:
I don't think you need to keep track of this. Just call `commit()` immediately as soon as you get an offset to commit. The commit policy will handle it.